autoPyTorch/pipeline/components/setup/network/backbone/tabular.py

```diff
@@ -0,0 +1,619 @@
+                                        min_num_gropus: int = 1,
+                                        max_num_groups: int = 9,
+                                        min_blocks_per_groups: int = 1,
+                                        max_blocks_per_groups: int = 4,
+                                        min_num_units: int = 10,
+                                        max_num_units: int = 1024,
+                                        ) -> ConfigurationSpace:
+        cs = ConfigurationSpace()
+
+        # Support for different shapes
+        resnet_shape = CategoricalHyperparameter(
+            'resnet_shape',
+            choices=[
+                'funnel',
+                'long_funnel',
+                'diamond',
+                'hexagon',
+                'brick',
+                'triangle',
+                'stairs'
+            ]
+        )
+        cs.add_hyperparameter(resnet_shape)
+
+        # The number of groups that will compose the resnet. That is,
+        # a group can have N Resblock. The M number of this N resblock
+        # repetitions is num_groups
+        num_groups = UniformIntegerHyperparameter(
+            "num_groups", lower=min_num_gropus, upper=max_num_groups, default_value=5)
+
+        blocks_per_group = UniformIntegerHyperparameter(
+            "blocks_per_group", lower=min_blocks_per_groups, upper=max_blocks_per_groups)
+
+        activation = CategoricalHyperparameter(
+            "activation", choices=list(_activations.keys())
+        )
+
+        output_dim = UniformIntegerHyperparameter(
+            "output_dim",
+            lower=min_num_units,
+            upper=max_num_units
+        )
+
+        cs.add_hyperparameters([num_groups, blocks_per_group, activation, output_dim])
+
+        # We can have dropout in the network for
+        # better generalization
+        use_dropout = CategoricalHyperparameter(
+            "use_dropout", choices=[True, False])
+        cs.add_hyperparameters([use_dropout])
+
+        use_shake_shake = CategoricalHyperparameter("use_shake_shake", choices=[True, False])
+        use_shake_drop = CategoricalHyperparameter("use_shake_drop", choices=[True, False])
+        shake_drop_prob = UniformFloatHyperparameter(
+            "max_shake_drop_probability", lower=0.0, upper=1.0)
+        cs.add_hyperparameters([use_shake_shake, use_shake_drop, shake_drop_prob])
+        cs.add_condition(CS.EqualsCondition(shake_drop_prob, use_shake_drop, True))
+
+        max_units = UniformIntegerHyperparameter(
+            "max_units",
+            lower=min_num_units,
+            upper=max_num_units,
+        )
+        cs.add_hyperparameters([max_units])
+
+        max_dropout = UniformFloatHyperparameter(
+            "max_dropout", lower=0.0, upper=1.0
+        )
+        cs.add_hyperparameters([max_dropout])
+        cs.add_condition(CS.EqualsCondition(max_dropout, use_dropout, True))
+
+        return cs
```

Comment:
We also want a DenseNet here at some point, perhaps even the one form AutoGluon (https://arxiv.org/pdf/2003.06505.pdf, Figure 1). Maybe add this to the issue about networks we want to still implement

Reply:
see #32